pkg/controllers/aws_infra.go

```diff
@@ -112,12 +112,15 @@ func (r *HypershiftDeploymentReconciler) createAWSInfra(hyd *hypdeployment.Hyper
 			}
 
 			hyd.Spec.HostedClusterSpec.IssuerURL = iamOut.IssuerURL
-			hyd.Spec.HostedClusterSpec.Platform.AWS.Roles = iamOut.Roles
+			hyd.Spec.HostedClusterSpec.Platform.AWS.RolesRef.ImageRegistryARN = iamOut.Roles.ImageRegistryARN
```

Comment:
will this affect the existing hypershiftDeployment?

Reply:
Good catch. The API change in the upstream hypershift around the roles and rolesRef seems backward incompatible.

Reply:
HD controller migrates roles to rolesRef